packages/headless/src/features/history/history-state.ts

```diff
@@ -18,28 +19,28 @@ export interface HistoryState extends SearchParametersState {
   facetOrder: string[];
 }
 
-function getSearchParametersInitialState(): SearchParametersState {
-  return {
-    context: getContextInitialState(),
-    facetSet: getFacetSetInitialState(),
-    dateFacetSet: getDateFacetSetInitialState(),
-    numericFacetSet: getNumericFacetSetInitialState(),
-    categoryFacetSet: getCategoryFacetSetInitialState(),
-    facetOptions: getFacetOptionsInitialState(),
-    pagination: getPaginationInitialState(),
-    query: getQueryInitialState(),
-    advancedSearchQueries: getAdvancedSearchQueriesInitialState(),
-    sortCriteria: getSortCriteriaInitialState(),
-    querySet: getQuerySetInitialState(),
-    pipeline: getPipelineInitialState(),
-    searchHub: getSearchHubInitialState(),
-    debug: getDebugInitialState(),
-  };
+export function getHistoryInitialState(): HistoryState {
+  return extractHistory({});
 }
 
-export function getHistoryInitialState(): HistoryState {
+export function extractHistory(state: Partial<HistoryState>): HistoryState {
   return {
-    ...getSearchParametersInitialState(),
-    facetOrder: [],
+    context: state.context || getContextInitialState(),
+    facetSet: state.facetSet || getFacetSetInitialState(),
+    numericFacetSet: state.numericFacetSet || getNumericFacetSetInitialState(),
+    dateFacetSet: state.dateFacetSet || getDateFacetSetInitialState(),
+    categoryFacetSet:
+      state.categoryFacetSet || getCategoryFacetSetInitialState(),
+    pagination: state.pagination || getPaginationInitialState(),
+    query: state.query || getQueryInitialState(),
+    advancedSearchQueries:
+      state.advancedSearchQueries || getAdvancedSearchQueriesInitialState(),
+    querySet: state.querySet || getQuerySetInitialState(),
+    sortCriteria: state.sortCriteria || getSortCriteriaInitialState(),
+    pipeline: state.pipeline || getPipelineInitialState(),
+    searchHub: state.searchHub || getSearchHubInitialState(),
+    facetOptions: state.facetOptions || getFacetOptionsInitialState(),
+    facetOrder: state.facetOrder ?? getFacetOrderInitialState(),
+    debug: state.debug ?? getDebugInitialState(),
```

Comment:
Could we destructure the `state` partial at the end to avoid writing all these conditionals? e.g.

```
return {
    ...getSearchParametersInitialState(),
    ...state,
}
```